demo/view-timeline/with-inset.html

```diff
@@ -0,0 +1,118 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>view-timeline demo</title>
+</head>
+<style type="text/css">
+  #container {
+    display:  flex;
+    flex-direction: column;
+    border:  1px solid black;
+    height:  500px;
+    width:  500px;
+    overflow-x:  hidden;
+    overflow-y:  scroll;
+    position:  relative;
+    align-items:  flex-start;
+  }
+
+  .overlay {
+    position: absolute;
+    top:  10px;
+    left:  10px;
+    width:  500px;
+    height:  500px;
+    pointer-events: none;
+  }
+
+  .spacer {
+    display:  inline-block;
+    flex:  none;
+    height:  120vh;
+    width:  120vw;
+  }
+
+  #subject {
+    background-color:  rgba(0, 0, 255, 0.5);
+    display:  inline-block;
+    flex:  none;
+    height: 100px;
+    width:  90%;
+  }
+
+  .progress-bar,
+  .progress-bar-progress {
+    border:  1px solid green;
+    height:  20px;
+    width:  200px;
+    position:  absolute;
+    inset-inline-start:  20px;
+    padding:  0;
+  }
+
+  .progress-bar > span {
+    padding-left:  5px;
+    padding-right:  5px;
+  }
+
+  .progress-bar-progress {
+    border-color:  transparent;
+    background-color: rgba(0, 200, 0, 0.3);
+    width: 0px;
+  }
+
+</style>
+<body>
+  <div id="container">
+    <div class="spacer"></div>
+    <div id="subject"></div>
+    <div class="spacer"></div>
+  </div>
+  <div class="overlay">
+    <div class="progress-bar" style="top: 20px;"><span>cover</span></div>
+    <div class="progress-bar" style="top: 70px;"><span>cover, inset 100px 200px</span></div>
+    <div class="progress-bar-progress" style="top: 20px;"></div>
+    <div class="progress-bar-progress" style="top: 70px;"></div>
+  </div>
+
+  <div class="overlay">
+    <div style="height: 99px"></div>
+    <div style="border-top: 1px solid red"></div>
+    <div style="height: 100px"></div>
+    <div style="height: 99px"></div>
+    <div style="border-top: 1px solid red"></div>
+  </div>
+
+</body>
+<script src="../../dist/scroll-timeline.js"></script>
+<script type="text/javascript">
+  "use strict";
+
+  const progressBars = document.querySelectorAll('.progress-bar-progress');
+  const createProgressAnimation = (bar, range, orientation, inset=null) => {
+    const subject = document.getElementById('subject');
+    const viewTimeline = new ViewTimeline({
+      'subject': subject,
+      'axis': orientation,
+      'inset': inset
+    });
+    bar.animate( { width: ['0px', '200px' ] }, {
+      timeline: viewTimeline,
+      timeRange: `${range}`,
```

Comment:
Suggest using delay and endDelay since timeRange is likely to be deprecated.  The current proposal is to change animation-delay to be a shorthand property which is start delay if one item is specified and start + end if two values are specified.